File: Java/SauceDemo/src/main/java/pages/InventoryPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class InventoryPage {
    WebDriver driver;

    By onesie_clothing_btn = By.id("add-to-cart-sauce-labs-onesie");
    By fleece_jacket_clothing_btn = By.id("add-to-cart-sauce-labs-fleece-jacket");
    By shopping_cart_btn = By.id("shopping_cart_container");
    By react_burger_menu_btn = By.id("react-burger-menu-btn");
    By logout_btn = By.id("logout_sidebar_link");
    public InventoryPage(WebDriver driver) {
        this.driver = driver;
    }

    public void clickOnesieClothingButton() {
        driver.findElement(onesie_clothing_btn).click();
    }

    public void clickFleeceJacketClothingButton() {
        driver.findElement(fleece_jacket_clothing_btn).click();
    }

    public void clickShoppingCartButton() {
        driver.findElement(shopping_cart_btn).click();
    }

    public void clickReactBurgerMenuButton() {
        driver.findElement(react_burger_menu_btn).click();
    }

    public void clickLogoutButton() {
        driver.findElement(logout_btn).click();
    }
}

File: Python/src/pages/inventory_page.py
"""
This module contains the elements of the inventory page
"""

from selenium.webdriver.common.by import By
from src.pages.side_bar import Sidebar

class InventoryPage(Sidebar):
    """
    This class contains the elements of the inventory page
    """
    def __init__(self, driver):
        super().__init__(driver)

        # Locators
        self.onesie_clothing_btn = By.ID, "add-to-cart-sauce-labs-onesie"
        self.fleece_jacket_clothing_btn = By.ID, "add-to-cart-sauce-labs-fleece-jacket"
        self.shopping_cart_btn = By.ID, "shopping_cart_container"

    def click_onesie_clothing_button(self):
        self.driver.find_element(*self.onesie_clothing_btn).click()

    def click_fleece_jacket_clothing_button(self):
        self.driver.find_element(*self.fleece_jacket_clothing_btn).click()

    def click_shopping_cart_button(self):
        self.driver.find_element(*self.shopping_cart_btn).click()
